### ISSUE TITLE
No longer works on 3.36

### ISSUE THREAD

**Issue by docquantum** [passingthru67/workspaces-to-dock#189]
###Linux Distribution version

Arch Linux 5.5.8-1-clear

###Gnome Shell version

GNOME Shell 3.36.0

###Xorg or Wayland (or both)

Wayland

###Extension version or branch

Current master

###Description of the problem

No longer loads nor works on 3.36

###Steps to reproduce the problem

Try to load the extension

**Comment by ysolis** [comment]
which error messages appears ? can you check your `journalctl` logs ?

**Comment by passingthru67** [comment]
@docquantum Working on it. I should have a Gnome 3.36 compatible version ready for testing soon.

**Comment by passingthru67** [comment]
@docquantum Gnome 3.36 Development branch uploaded at https://github.com/passingthru67/workspaces-to-dock/tree/gnome-3.36-dev.
Please download and install manually. If there are no major issues, I'll merge into master and publish to the extensions.gnome.org site.

**Comment by ysolis** [comment]
@passingthru67 works for me :)

**Comment by vivaeltopo** [comment]
can confirm, works.

**Comment by OrderlyUnicode** [comment]
I was having problems loading it, but I think I installed it wrong.

**Comment by dpriedel** [comment]
work for me too

thanks !!!

**Comment by iJackUA** [comment]
Works on Gnome 3.36, please merge it

**Comment by xakraz** [comment]
Works for me too for Gnome 3.36 on Manjaro !

:pray: Thanks for your work ! 
This extension is a must on Gnome and should be part of the built-in ones !

Keep-up the good work !
Thanks again !

:clap: :clap: :clap: :clap: :clap:

**Comment by mwildam** [comment]
Yes, indeed, if this extension does not work would be a showstopper for any
migration! Could not find a way to donate.

Best regards, Martin.

**Comment by vivaeltopo** [comment]
@passingthru67 deprecated warning upon shell restart:

`Mar 28 11:53:24 deepthought gnome-shell[11976]: Usage of object.actor is deprecated for WorkspacesToDock_ShortcutButton
                                                get@resource:///org/gnome/shell/ui/environment.js:285:29
                                                _createIcon@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:262:13
                                                _createIconTexture@resource:///org/gnome/shell/ui/iconGrid.js:104:26
                                                setIconSize@resource:///org/gnome/shell/ui/iconGrid.js:97:14
                                                _init@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:153:19
                                                _populate@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:1292:38
                                                WorkspacesToDock_ShortcutsPanel@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:951:14
                                                WorkspacesToDock_DockedWorkspaces@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:298:32
                                                enable@/usr/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/extension.js:97:12
                                                _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:160:32
                                                loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:311:26
                                                _loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:536:18
                                                collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:17
                                                _loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:515:19
                                                _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:545:18
                                                _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:576:18
                                                init@resource:///org/gnome/shell/ui/extensionSystem.js:54:14
                                                _initializeUI@resource:///org/gnome/shell/ui/main.js:249:22
                                                start@resource:///org/gnome/shell/ui/main.js:138:5
                                                @<main>:1:47
`

**Comment by docquantum** [comment]
I get the same error

```
Usage of object.actor is deprecated for WorkspacesToDock_ShortcutButton
                                                 get@resource:///org/gnome/shell/ui/environment.js:285:29
                                                 _createIcon@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:262:13
                                                 _createIconTexture@resource:///org/gnome/shell/ui/iconGrid.js:104:26
                                                 setIconSize@resource:///org/gnome/shell/ui/iconGrid.js:97:14
                                                 _init@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:153:19
                                                 _populate@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:1292:38
                                                 WorkspacesToDock_ShortcutsPanel@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/shortcutsPanel.js:951:14
                                                 WorkspacesToDock_DockedWorkspaces@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:298:32
                                                 enable@/home/dshchur/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/extension.js:97:12
                                                 _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:160:32
                                                 _onEnabledExtensionsChanged/<@resource:///org/gnome/shell/ui/extensionSystem.js:446:18
                                                 _onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:445:11
                                                 _init/createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:529:46
                                                 enableExtension@resource:///org/gnome/shell/ui/extensionSystem.js:187:29
                                                 EnableExtension@resource:///org/gnome/shell/ui/shellDBus.js:307:38
                                                 _handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:327:38
                                                 _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:404:34
```

Thanks for the work passingthru, it seems to work now on the new branch.

**Comment by smcv** [comment]
(For context: I don't use this extension myself, but I'm going through GNOME Shell extensions that are packaged in Debian and making sure there are issues open to get them adapted to an upstream change.)

One thing that isn't going to work in the gnome-3.36-dev branch: in a couple of places, this extension uses

```
Util.spawn(["gnome-shell-extension-prefs", Me.metadata.uuid]);
```

which will not work since GNOME Shell 3.36.1 (the way extension preferences work has been redone to accommodate the g-s-extension-prefs app being installed via Flatpak or similar).
                                                                                  
As far as I can tell from GNOME Shell's upstream commit history, the preferred way to launch extension preferences in sufficiently recent versions is to call imports.misc.extensionUtils.openPrefs(), which was added by [gnome-shell!1163](https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1163). The version in Ubuntu 20.04 doesn't have that function available yet, but hopefully will soon; Debian experimental already has it.
                                                                                  
To support older GNOME versions, fall back to the spawn call if imports.misc.extensionUtils doesn't have an openPrefs method.
                                                                                  
Sample code: [Tudmotu/gnome-shell-extension-clipboard-indicator#203](https://github.com/Tudmotu/gnome-shell-extension-clipboard-indicator/pull/203)
